opm/autodiff/WellHelpers.hpp

```diff
@@ -191,6 +191,63 @@ namespace Opm {
             return retval;
         }
 
+        template <class MatrixType>
+        bool invertMatrixLapack(MatrixType& matrix) {
+
+            const int m = matrix.M();
+            const int n = matrix.N();
+            assert(m == n); // it needs to be a square matrix
+
+            // the matrix
+            double* A = new double[m * n];
+
+            // copying the data in matrix to A
+            int i_count = 0;
+            for (int irow = 0; irow < m; ++irow) {
+                for (int icol = 0; icol < n; ++icol) {
+                    A[i_count] = matrix[irow][icol];
+                    ++i_count;
```

Comment:
This loop nest implicitly assumes that the `matrix` is symmetric.  Is that the case?  If not, then it's better to rewrite it as
```C++
for (row) {
    for (col) {
        A[row + col*n] = matrix[row][col];
    }
}
```

Reply:
> This loop nest implicitly assumes that the matrix is symmetric

It is not assuming the matrix is symmetric, while we got the transpose and get the inverse of the transpose in the following and then transpose again when putting back to matrix. Sorry for the confusion. It was not intended at  the beginning while discovered later.